### ISSUE TITLE
function AppWeb.Router.Helpers.live_path/2 is undefined or private

### ISSUE THREAD

**Issue by aesmail** [phoenixframework/phoenix#3771]
### Environment

* Elixir version (elixir -v): 1.10.2
* Phoenix version (mix deps): 1.5.0-rc.0
* NodeJS version (node -v): v8.16.1
* NPM version (npm -v): 6.4.1
* Operating system: macOS Catalina 10.15.4

### Expected behavior
On a fresh `mix phx.new app --live` application, there's a `PageLive` LiveView page with a route:
`live "/", PageLive, :index`

Doing `<%= live_patch "My Page", to: Routes.live_path(@socket, PageLive) %>` should take me to that page or at least doesn't raise an `undefined` error.

### Actual behavior
Having `<%= live_patch "My Page", to: Routes.live_path(@socket, PageLive) %>` anywhere in the template raises an `UndefinedFunctionError` no matter whether I have `live` routes or not.

**Comment by nickjj** [comment]
I'm able to reproduce this too.

What I ended up doing to fix it for now was remove the `:index` from the live route in the `router.ex` and also reference `HelloWeb.PageLive` in the `live_patch` instead of `PageLive`.

I'm not sure if this is the best way to work around these issues, or if they are oversights in the generated app's code.

**Comment by aesmail** [comment]
@nickjj you're right.

**RAISES EXCEPTIONS: 1**

```elixir
# router.ex
scope "/", HelloWeb do
    pipe_through :browser
    live "/", PageLive, :index
end
```
```elixir
# page_live.html.leex
# all the following lines raise the exception
<%= live_patch "Home", to: Routes.live_path(@socket, PageLive, :index) %>
<%= live_patch "Home", to: Routes.live_path(@socket, HelloWeb.PageLive, :index) %>
<%= live_patch "Home", to: Routes.live_path(@socket, HelloWeb.PageLive) %>
<%= live_patch "Home", to: Routes.live_path(@socket, PageLive) %>
```

**RAISES EXCEPTIONS: 2**

```elixir
# router.ex
scope "/", HelloWeb do
    pipe_through :browser
    live "/", PageLive # removed action from route
end
```

```elixir
# page_live.html.leex

# the following lines raise different exceptions
<%= live_patch "Home", to: Routes.live_path(@socket, PageLive) %>
<%= live_patch "Home", to: Routes.live_path(@socket, PageLive, :index) %>
<%= live_patch "Home", to: Routes.live_path(@socket, HelloWeb.PageLive, :index) %>
```

**WORKS AS EXPECTED**

```elixir
# router.ex
scope "/", HelloWeb do
    pipe_through :browser
    live "/", PageLive # removed action from route
end
```

```elixir
# page_live.html.leex

# the following lines work as expected
<%= live_patch "Home", to: Routes.live_path(@socket, HelloWeb.PageLive) %>
<%= live_patch "Home", to: Routes.live_path(@socket, HelloWeb.PageLive, page: 4) %>
```

**Comment by chrismccord** [comment]
You need `Routes.page_path(@socket, :index)`. When an action is provided to the `live` routes, we generate named routes + actions the same way controller routes are generate. Check `mix phx.routes`. Thanks!

**Comment by plicjo** [comment]
@nickjj @aesmail

I also was able to reproduce this problem.

Under no combination of arity (I tried 0/1/2/3/4) was I able to get the following `live_patch` with the page based link to render.

I was typing up a deep response to this post...examining every character and every line....and to my frustration IT JUST STARTED WORKING!

The error I was getting was:
```
function LocalFareJaxWeb.Router.Helpers.user_index_path/2 is undefined or private
```

As this is dynamically generated (I think) by what is in the router, my only conclusion is that there is something going undetected during recompilation.

As a result, my response to you guys and my future self would be:

1. Make sure you've saved the router file (haha).
2. Restart the phoenix server for sanity's sake.

And it should (?) work. This was entirely baffling.
